ecs-agent/api/ecs/client/ecs_client_test.go

```diff
@@ -386,6 +386,8 @@ func TestRegisterContainerInstance(t *testing.T) {
 						Return("", errors.New("fake unit test error")),
 					mockEC2Metadata.EXPECT().GetDynamicData(ec2.InstanceIdentityDocumentResource).
 						Return(expectedIID, nil),
+					mockEC2Metadata.EXPECT().GetDynamicData(ec2.InstanceIdentityDocumentSignatureResource).
+						Return("", errors.New("fake unit test error")),
```

Comment:
Why is this change needed? IIUC other than retries the behavior remains the same?

Reply:
Thanks for this comment. Firstly (separate from the question at hand), it seems L383-L393 is never getting executed at the moment as L383 is currently never true.. I will address this bug in the next revision. 

Regarding the original question: assuming that L383-L393 DOES get executed, the change is to test the case of getting instance identity signature failing with some error and making sure that we do retry in such a case. The test now expects Agent to first try to retrieve the instance identity signature, mocks an error to be returned from doing so, and then expects Agent to try to retrieve the instance identity signature once again upon encountering that aforementioned error. Without the new retry backoff logic introduced to `ecs-agent/api/ecs/client/ecs_client.go`, this modified test would fail since it now requires that retrieving the instance identity signature is retried on errors.

Reply:
Ah I didn't notice the expectation added on L391. LGTM.